### PR TITLE
docs(service-defaults): upstream overrides for peered services

### PR DESCRIPTION
**Pull request by DanStough**
### Description
Docs for ##15956. Upstream overrides will require the `peer` field to work for peered services in 1.16.

### CODE REVIEW

File: website/content/docs/connect/config-entries/mesh.mdx
Comment (DanStough) [inline website/content/docs/connect/config-entries/mesh.mdx:329]: No idea why, but these links are messed up in prod. Though maybe removing the prefix would help.

**Review by hashi-derek — APPROVED** [review]
This looks really good. Thanks for taking the time to polish this. I'm approving this, but we might wanna wait for someone from the docs team to also approve before merging.

File: website/content/docs/connect/config-entries/service-defaults.mdx
Comment (trujillo-adam) [inline website/content/docs/connect/config-entries/service-defaults.mdx:432]:
```suggestion
                `The name of the peer containing the upstream. Do not use a wildcard specifier ( \`*\`).<br><br>
                    If the \`peer\` field is not set in an \`Overrides\` configuration, then Consul applies overrides to peered services with the same [\`name\`](/consul/docs/connect/config-entries/service-defaults#name),
                    even if the \`peer\` field is set in the service configuration. This maintains backward compatibility with Consul 1.14.0. 
                    If the \`peer\` field is specified in any override, then backward compatibility with Consul 1.14.0 is disabled. As a result, each peered service must have an override configuration.` ,
```
A few suggestions to switch to active voice and to remove temporal language per the style guide. Also, I was a little confused by the "even if this field is set" part of the NOTE statement. It reads like contradictory statements. I tried to add something that made sense, but I don't think it's correct. Do we mean that the peer field must be set for all overrides, otherwise the override configuration is applied to any service name matching the configuration in any peered cluster?

File: website/content/docs/upgrading/upgrade-specific.mdx
Comment (trujillo-adam) [inline website/content/docs/upgrading/upgrade-specific.mdx:92]:
```suggestion
In Consul 1.14.x, `service-defaults` upstream [`overrides`](/consul/docs/connect/config-entries/service-defaults#overrides) applies to both local and peered services as long as the `name` field matched.
```

File: website/content/docs/upgrading/upgrade-specific.mdx
Comment (trujillo-adam) [inline website/content/docs/upgrading/upgrade-specific.mdx:93]:
```suggestion
Consul 1.15.0 is backward compatible with 1.14 if the [`peer`](/consul/docs/connect/config-entries/service-defaults#peer) is not set in any override.
```
Just to make this simpler and remove the suggestion that we're implementing a new mode that the user has to keep track of. I understand this just to be more of a behavior

File: website/content/docs/upgrading/upgrade-specific.mdx
Comment (trujillo-adam) [inline website/content/docs/upgrading/upgrade-specific.mdx:94]:
```suggestion
```
I recommend removing future-oriented language. I would reinsert this or a similar statement when 1.16 is available.

**Review by trujillo-adam — COMMENTED** [review]
Added some suggestions for style consistency and a few questions to help improve the clarity.

File: website/content/docs/connect/config-entries/service-defaults.mdx
Comment (DanStough) [reply website/content/docs/connect/config-entries/service-defaults.mdx:432]: Actually I think this is pretty close to what I was trying to say, I think we just need to remove `even if the \`peer\` field is set in the service configuration` for clarity.

**Review by trujillo-adam — APPROVED** [review]
looks great 👍

**Pull request merged**
